api/src/app/modules/project/project.service.ts

```diff
@@ -503,6 +507,27 @@ export class ProjectService extends DataService<Project, Repository<Project>, Pr
         throw new BadRequestException(`Unable to transition FOM ${entity.id} to ${stateTransition}.  
         Proposed submission is required.`);
       }
+
+      // Checks postDate on or before commenting open date and 1 day later than today (before published)
+      // However, it could be empty = then no check.
+      const publicNotices = entity.publicNotices;
+      if (!_.isEmpty(publicNotices)) {
+        const postDate = publicNotices[0].postDate;
+        const commentingOpenDate = entity.commentingOpenDate;
+        if (postDate && !isPNPostdateOnOrBeforeCommentingOpenDate(postDate, commentingOpenDate)) {
+			throw new BadRequestException(`Unable to transition FOM ${entity.id} to ${stateTransition}. 
+            Online Public Notice post date ${postDate} should be on or before commenting start date 
+            ${commentingOpenDate}.`);
+        }
+
+        // Must be at least one day after publish is pushed
+        const dayDiff = DateTimeUtil.diffNow(postDate, DateTimeUtil.TIMEZONE_VANCOUVER, 'day');
```

Comment:
What if postDate is null here?

Reply:
**Good point**, but strange things is I ran flow from "admin" ui and still works fine when postDate is null and final outcome seem to be correct. Strange, somewhere if it is null probably already skipped this part of validation. I will check why I didn't catch this. Thanks.

Reply:
The intention initially was, if postDate is empty, no need to do validation (since it will be set to default commenting open date when FOM state changes to PUBLISH).

Ha, I checked why the flow was still good, turns out even if date is null, the datejs library returns a null for the "diff". So, accidently, the next line is still evaluated to be false so the validation pass, leading that I didn't detect this bug when running different scenarios using frontend flow. Anyway, I added a condition in. Thanks for catching this.